@here/harp-mapview/lib/MapView.ts

```diff
@@ -2539,9 +2544,17 @@ export class MapView extends THREE.EventDispatcher {
 
         const renderList = this.m_visibleTiles.dataSourceTileList;
 
+        const hasTextRenderer = this.m_textElementsRenderer !== undefined;
+
+        // no need to check everything if we're not going to create it.
+        let hasTextToRender = hasTextRenderer || !this.renderLabels;
```

Comment:
Not really. If we set .renderLabels = false, and then set theme, text renderer would be destroyed and won't created again. So this would work.

Reply:
Ok, i get the shortcut but it's heavily misleading because of variable name.

O, so maybe i misunderstand because name says "has text to render", but in reality it can be `true` in some weird circumstances like:
 - no text, but textRenderer is already created 
 - no text, but renderLabels = false
When name says  its true if
 - there is text on some tile

So, i see problem with naming (inaccurate at best) and convoluted semantics (still relying on other variables). 

Do we really want to name it "hasTextToRenderOrTileCheckNotNeeded" ... or just `flag` with big comment.

Reply:
Yep, it's confusing a bit. Rewrited it in more clear way.